include/util/hex_float.h

```diff
@@ -39,19 +39,43 @@
 
 namespace spvutils {
 
+class Float16 {
+ public:
+  Float16(uint16_t v): val(v) {}
+  Float16() {}
+  static bool isNan(const Float16 val) {
+    return (val.val & 0x7FFF) == 0x7C00;
```

Comment:
I think that's the is-infinity test.
isnan should be 
(val.val & 0x7C00 == 0x7C00) && (val.val & 3FF != 0)